src/libbson/src/bson/bson-atomic.h

```diff
@@ -32,6 +32,8 @@
 
 BSON_BEGIN_DECLS
 
```

Comment:
Since `bson-atomic.h` is the only header that requires `MC_DISABLE_COVERED_SWITCH_DEFAULT_*` and `MC_DISABLE_ATOMIC_IMPLICIT_SEQ_CST_*`, suggest defining them (and undefining) within this file:

```c
// ... #define macros ...

// ... rest of header ...

// ... #undef macros ...
```

That removes the macros from public API. `bson-atomic.h` is planned for removal in 2.0 (CDRIVER-5703).